resolve Error: "pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior."
define BOOST_BIND_GLOBAL_PLACEHOLDERS
replace #include <boost/bind.hpp> with #include <boost/bind/bind.hpp>
diff -ru Dice.orig/ZSHCI/Hmult.h Dice/ZSHCI/Hmult.h
--- Dice.orig/ZSHCI/Hmult.h	2024-01-11 17:10:03.915355000 +0100
+++ Dice/ZSHCI/Hmult.h	2024-01-11 17:20:04.223600142 +0100
@@ -32,6 +32,7 @@
 #include <chrono>
 #include "SHCISortMpiUtils.h"
 #include "SHCImake4cHamiltonian.h"
+#define BOOST_BIND_GLOBAL_PLACEHOLDERS
 
 using namespace Eigen;
 using namespace std;
diff -ru Dice.orig/ZSHCI/iowrapper.h Dice/ZSHCI/iowrapper.h
--- Dice.orig/ZSHCI/iowrapper.h	2024-01-11 17:10:03.926065371 +0100
+++ Dice/ZSHCI/iowrapper.h	2024-01-11 17:50:17.289606580 +0100
@@ -18,6 +18,7 @@
 */
 #ifndef IOWRAPPER_HEADER_H
 #define IOWRAPPER_HEADER_H
+#define BOOST_BIND_GLOBAL_PLACEHOLDERS
 #include <Eigen/Dense>
 #include <boost/serialization/serialization.hpp>
 #include <boost/serialization/complex.hpp>
diff -ru Dice.orig/ZSHCI/SHCI.cpp Dice/ZSHCI/SHCI.cpp
--- Dice.orig/ZSHCI/SHCI.cpp	2024-01-11 17:10:03.917251803 +0100
+++ Dice/ZSHCI/SHCI.cpp	2024-01-11 17:19:30.845898274 +0100
@@ -59,7 +59,7 @@
 #include "symmetry.h"
 MatrixXd symmetry::product_table;
 #include <algorithm>
-#include <boost/bind.hpp>
+#include <boost/bind/bind.hpp>
 
 // Initialize
 using namespace Eigen;
diff -ru Dice.orig/ZSHCI/symmetry.cpp Dice/ZSHCI/symmetry.cpp
--- Dice.orig/ZSHCI/symmetry.cpp	2024-01-11 17:10:03.927561000 +0100
+++ Dice/ZSHCI/symmetry.cpp	2024-01-11 17:38:01.888632035 +0100
@@ -25,7 +25,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <algorithm>
-#include <boost/bind.hpp>
+#include <boost/bind/bind.hpp>
 #include <Eigen/Dense>
 
 #include "integral.h"
